src/main/java/gift/web/ProductController.java

```diff
@@ -0,0 +1,57 @@
+package gift.web;
+
+import gift.web.dto.Product;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController //컨트롤러를 JSON을 반환하는 컨트롤러로 만들어줌
+@RequestMapping("/api/products")
+public class ProductController {
+
+    private final Map<Long, Product> products = new HashMap<>();
+
+    @GetMapping
+    public ResponseEntity<List<Product>> getProducts() {
+        // 여기에는 products 전체 상품을 리턴
+        return new ResponseEntity<>(List.copyOf(products.values()), HttpStatus.OK);
+    }
+
+    // products/{상품번호}의 GetMapping
+    @GetMapping("/{id}") //Query Param과 Path Variable 사용의 차이점 알아보기
+    public ResponseEntity<Product> getProduct(@PathVariable Long id) {
+        if(products.containsKey(id)) return new ResponseEntity<>(products.get(id), HttpStatus.OK);
```

Comment:
코드 전체적으로 스타일 가이드를 지킬 수 있게 수정해볼까요?
```suggestion
       if (products.containsKey(id)) {
            return new ResponseEntity<>(products.get(id), HttpStatus.OK);
        }
```